### PR TITLE
[FEAT #119] markdown live preview

### PR DESCRIPTION
**Pull request by fourplusone**
Here's my approach to tackle #119: If a mode contains a resource called `WebPreview.js` the script will be used to transform the text to HTML by calling `webPreview(text) -> string`

An example is implemented in the `Markdown.seemode` using showdown.js

Things to be done before making this available as public API:

- [x] error handling & reporting (right now it just fails silently)
- [x] settle on naming
- [x] document this feature: https://github.com/subethaedit/SubEthaEdit-Modes/pull/6
- [x] move the webPreview work on a background queue

### CODE REVIEW

**Comment by monkeydom** [comment]
Hmm. Seems a simple enough approach. And with all the tooling around .js it might very well be possible to compile a lot of the functionality we'd like to see down to a working standalong static .js file. I like it.

**Comment by monkeydom** [comment]
Eventually I do think we want more for this (e.g. persistent options for the preview, etc). But for now I think it is a good start and works rather nicely. I did add a github style template and added some metadata. Please double check again.

File: SubEthaEdit-Mac/Modes/Markdown.seemode/Contents/Resources/WebPreview.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
...r+"</u>":e})).replace(/__(\S[\s\S]*?)__/g,function(e,r){return/\S$/.test(r)?"<u>"+r+"</u>":e}),e=e.replace(/(_)/g,a.helper.escapeCharactersCallback),e=t.converter._dispatch("underline.after",e,r,t)):e}),a.subParser("unescapeSpecialChars",function(e,r,t){"use strict";return e=t.converter._dispatch("unescapeSpecialChars.before",e,r,t),e=e.replace(/¨E(\d+)E/g,function(e,r){var t=parseInt(r);return String.fromCharCode(t)}),e=t.converter._dispatch("unescapeSpecialChars.after",e,r,t)}),a.subParser("makeMarkdown.blockquote",function(e,r){"use strict";var t="";if(e.hasChildNodes())for(var n=e.childNodes,s=n.length,o=0;o<s;++o){var i=a.subParser("makeMarkdown.node")(n[o],r);""!==i&&(t+=i)}return t=t.trim(),t="> "+t.split("\n").join("\n> ")}),a.subParser("makeMarkdown.codeBlock",function(e,r){"use strict";var t=e.getAttribute("language"),a=e.getAttribute("precodenum");return"```"+t+"\n"+r.preList[a]+"\n```"}),a.subParser("makeMarkdown.codeSpan",function(e){"use strict";return"`"+e.innerHTML+"`"}),a.subParser("makeMarkdown.emphasis",function(e,r){"use strict";var t="";if(e.hasChildNodes()){t+="*";for(var n=e.childNodes,s=n.length,o=0;o<s;++o)t+=a.subParser("makeMarkdown.node")(n[o],r);t+="*"}
```
Comment (fourplusone) [inline SubEthaEdit-Mac/Modes/Markdown.seemode/Contents/Resources/WebPreview.js]: I think this function is unused

**Comment by fourplusone** [comment]
Looks good to me

File: SubEthaEdit-Mac/Modes/Markdown.seemode/Contents/Resources/WebPreview.js (hunk 1 above)
Comment (monkeydom) [reply SubEthaEdit-Mac/Modes/Markdown.seemode/Contents/Resources/WebPreview.js]: If you refer to escapeHTML then yes, you are right, But I wanted to leave that in there for debugging purposes.

**Pull request merged**